core/tests/puppeteer-acceptance-tests/utilities/user/translation-admin.ts

```diff
@@ -118,19 +142,30 @@ export class TranslationAdmin extends BaseUser {
   async viewContributorTranslationRightsByLanguageCode(
     languageCode: string
   ): Promise<void> {
+    // Wait for the contributor filter method dropdown to be available before selecting a filter.
+    await this.page.waitForSelector(viewContributorFilterMethodSelect);
+    // Wait for the submit button to be visible before attempting to click it.
+    await this.page.waitForSelector(viewContributorSubmitButton);
+
     await this.select(viewContributorFilterMethodSelect, roleMethodValue);
     await this.select(viewContributorCategorySelect, translationRightValue);
     await this.select(viewContributorLanguageSelect, languageCode);
     await this.clickOn(viewContributorSubmitButton);
-
+    // Wait for the language role user result element to appear after filtering users.
+    await this.page.waitForSelector(viewLanguageRoleUserResult);
     await this.waitForNetworkIdle();
   }
 
   /**
    * Function to check if the language is displayed as a translation right.
    */
   async expectDisplayedLanguagesToContain(language: string): Promise<void> {
-    await this.page.waitForSelector(viewContributorLanguageResult);
+    const elementHandle = await this.page.$(viewLanguageRoleUserResult);
+    if (!elementHandle) {
+      showMessage(
+        'User list element (.e2e-test-reviewer-roles-result) not found assuming no users have rights.'
+      );
+    }
```

Comment:
why this change? why is this better than waiting for the `viewContributorLanguageResult` selector?